File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/GetAllRoomOptionsResponse.java
package com.ll.hotel.domain.hotel.room.dto;

import com.ll.hotel.domain.hotel.option.entity.RoomOption;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public record GetAllRoomOptionsResponse(
        Set<String> roomOptions
) {
    public GetAllRoomOptionsResponse(List<RoomOption> roomOptions) {
        this(
                roomOptions.stream()
                        .map(RoomOption::getName)
                        .collect(Collectors.toSet())
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/GetAllRoomOptionsResponse.kt
package com.ll.hotel.domain.hotel.room.dto

import com.ll.hotel.domain.hotel.option.entity.RoomOption

data class GetAllRoomOptionsResponse(
    val roomOptions: Set<String>
) {
    constructor(roomOptions: List<RoomOption>) : this(
        roomOptions.map { it.name }.toSet()
    )
}